docs/core/howto/ssl.rst

```diff
@@ -204,10 +199,40 @@ For example,
     options = CertificateOptions(..., dhParameters=dhParams)
 
 Another part of the TLS protocol which ``CertificateOptions`` can control is the version of the TLS or SSL protocol used.
-This is often called the context's "method".
+There are two ways to configure it, either by configuring a list of protocols, or by the legacy way of setting the internal context's "method".
+By default, Twisted will configure it to use TLSv1.0 or later and disable the insecure SSLv3 protocol, but this manual control over protocols can be helpful if you need to support legacy SSLv3 systems, or you wish to restrict it down to just the strongest of the TLS versions.
+
+You can give ``CertificateOptions`` a range of TLS versions you wish it to negotiate with the ``tlsProtocols`` argument in the initializer:
+
+.. code-block:: python
+
+    from twisted.internet.ssl import (CertificateOptions, TLSVersion,
+                                      TLSVersionRange)
+    options = CertificateOptions(
+        ...,
+        tlsProtocols=TLSVersionRange(TLSVersion.TLSv1_0,
+                                     TLSVersion.TLSv1_2))
+
```

Comment:
To expand on this a bit: there is essentially zero utility in being able to exclude a TLS version in the middle of your supported range. For example, if you want to support TLSv1 and TLSv1.2, there is no good reason to exclude TLSv1.1: clients that support TLSv1.2 will still use it, and clients that don't support TLSv1.2 will still have the option of choosing TLSv1.1 if they want to (they won't for various historical reasons, but that option exists).

In that case, using a set-based approach is just a really substantial increase in redundancy as well as an opportunity to do stupid stuff. On *top* of that, it means that if you have an application that you wrote in 2016 that is still being used in 2024 when a hypothetical TLSv1.4 is used, the set of "included only" TLS versions would *exclude* the use of TLSv1.4. That's a really silly limitation, so any implementation must either be a blacklist of TLS versions or must include some way to encode the notion of "X version *or better*".

I'm not wedded to the range, and alternative implementations that meet those goals would be fine by me, but I think the set is a bad choice for those reasons.